var/spack/repos/builtin/packages/nek5000/package.py

```diff
@@ -76,27 +81,106 @@ class Nek5000(Package):
 
     # Dependencies
     depends_on('mpi', when="+mpi")
+    depends_on('libx11', when="+prenek")
+    depends_on('libx11', when="+postnek")
+    # libxt is needed for X11/Intrinsic.h but not for linking
+    depends_on('libxt', when="+prenek")
+    depends_on('libxt', when="+postnek")
     depends_on('visit', when="+visit")
 
     @run_before('install')
     def fortran_check(self):
-        if not self.compiler.fc:
-            msg = 'Cannot build Nek5000 without a Fortran compiler.'
+        if not self.compiler.f77:
+            msg = 'Cannot build Nek5000 without a Fortran 77 compiler.'
+            raise RuntimeError(msg)
+
+    @run_after('install')
+    def test_install(self):
+        currentDir = os.getcwd()
+        eddyDir = 'short_tests/eddy'
+        os.chdir(eddyDir)
+
+        os.system(join_path(self.prefix.bin, 'makenek') + ' eddy_uv')
+        if not os.path.isfile(join_path(os.getcwd(), 'nek5000')):
+            msg = 'Cannot build example: short_tests/eddy.'
             raise RuntimeError(msg)
 
+        os.chdir(currentDir)
+
     def install(self, spec, prefix):
         toolsDir   = 'tools'
         binDir     = 'bin'
 
-        FC  = self.compiler.fc
+        # Do not use the Spack compiler wrappers.
+        # Use directly the compilers:
+        FC  = self.compiler.f77
```

Comment:
Is there a reason Nek doesn’t use the wrappers?  Do they actually break something, and does it RPATH its dependencies itself?

Reply:
I don't exactly remember what the problem I was having with compiler wrappers. I didn't
do a great job documenting it either. I will just keep this as is since we tested with these.
Nek5000 does not depend on any other spack package.

Reply:
For next time can you migrate these to use the wrappers?  I’m curious whether Nek will be able to find its dependencies and run in all cases.  Does the build RPATH them?

Reply:
I don't think Nek5000 RPATH the dependencies.

We could build Nek5000 on almost all the places we tested except in `summit-dev`, I think.

Reply:
Nek5000 only depends `mpi` and on libx11 if the tools `prenek` and `postnek` are built. It is pretty much self contained.

Reply:
Ok just note that if x11 is not system x11 or if the system MPI doesn’t RPATH properly, it may fail to run.

Reply:
@tgamblin, the reason I explicitly put the compilers and not the wrappers is that Nek needs these outside the Spack installation where the Spack compiler wrapper will not work properly since it requires proper package-specific environment.